--- libvirt-0.10.2.4/src/libxl/libxl_conf.c.orig	2013-04-26 16:14:27.233001065 +0000
+++ libvirt-0.10.2.4/src/libxl/libxl_conf.c	2013-04-26 16:16:10.456001296 +0000
@@ -542,27 +542,7 @@
             x_disk->backend = LIBXL_DISK_BACKEND_PHY;
         } else if (STREQ(l_disk->driverName, "qemu")) {
             x_disk->backend = LIBXL_DISK_BACKEND_QDISK;
-            switch (l_disk->format) {
-            case VIR_STORAGE_FILE_QCOW:
-                x_disk->format = LIBXL_DISK_FORMAT_QCOW;
-                break;
-            case VIR_STORAGE_FILE_QCOW2:
-                x_disk->format = LIBXL_DISK_FORMAT_QCOW2;
-                break;
-            case VIR_STORAGE_FILE_VHD:
-                x_disk->format = LIBXL_DISK_FORMAT_VHD;
-                break;
-            case VIR_STORAGE_FILE_NONE:
-                /* No subtype specified, default to raw */
-            case VIR_STORAGE_FILE_RAW:
-                x_disk->format = LIBXL_DISK_FORMAT_RAW;
-                break;
-            default:
-                virReportError(VIR_ERR_INTERNAL_ERROR,
-                               _("libxenlight does not support disk format %s"),
-                               virStorageFileFormatTypeToString(l_disk->format));
-                return -1;
-            }
+            x_disk->format = LIBXL_DISK_FORMAT_RAW;
         } else {
             virReportError(VIR_ERR_INTERNAL_ERROR,
                            _("libxenlight does not support disk driver %s"),
