diff --git a/Makefile b/Makefile
index ee70aa8..6d6a461 100644
--- a/Makefile
+++ b/Makefile
@@ -8,7 +8,7 @@ VPATH=src examples
 BUILDDIR=_build
 PROJECTS=configure configured ctypes ctypes-foreign-base ctypes-foreign-threaded ctypes-foreign-unthreaded ctypes-top
 GENERATED=src/ctypes_config.h src/ctypes_config.ml setup.data src/ctypes/ctypes_primitives.ml
-CFLAGS=-fPIC -Wall -O3 $(OCAML_FFI_INCOPTS)
+CFLAGS=-std=gnu99 -fPIC -Wall -O3 $(OCAML_FFI_INCOPTS)
 OCAML_FFI_INCOPTS=$(libffi_opt)
 
 # public targets
