diff --git a/drivers/SR.py b/drivers/SR.py
index 41c6ce7..c8ec2c8 100755
--- a/drivers/SR.py
+++ b/drivers/SR.py
@@ -27,7 +27,7 @@ import copy, os
 
 MOUNT_BASE = '/var/run/sr-mount'
 DEFAULT_TAP = 'vhd'
-TAPDISK_UTIL = '/usr/sbin/td-util'
+TAPDISK_UTIL = '@LIBDIR@/blktap/sbin/td-util'
 MASTER_LVM_CONF = '/etc/lvm/master'
 
 # LUN per VDI key for XenCenter
diff --git a/drivers/blktap2.py b/drivers/blktap2.py
index a0a25b3..f3029f7 100755
--- a/drivers/blktap2.py
+++ b/drivers/blktap2.py
@@ -135,7 +135,7 @@ def retried(**args): return RetryLoop(**args)
 class TapCtl(object):
     """Tapdisk IPC utility calls."""
 
-    PATH = "/usr/sbin/tap-ctl"
+    PATH = "@LIBDIR@/blktap/sbin/tap-ctl"
 
     def __init__(self, cmd, p):
         self.cmd    = cmd
diff --git a/drivers/vhdutil.py b/drivers/vhdutil.py
index 94696f7..ab6d444 100755
--- a/drivers/vhdutil.py
+++ b/drivers/vhdutil.py
@@ -28,7 +28,7 @@ import re
 
 MAX_VHD_JOURNAL_SIZE = 6 * 1024 * 1024 # 2MB VHD block size, max 2TB VHD size
 MAX_CHAIN_SIZE = 30 # max VHD parent chain size
-VHD_UTIL = "/usr/bin/vhd-util"
+VHD_UTIL = "@LIBDIR@/blktap/bin/vhd-util"
 OPT_LOG_ERR = "--debug"
 VHD_BLOCK_SIZE = 2 * 1024 * 1024
 VHD_FOOTER_SIZE = 512
@@ -324,7 +324,7 @@ def _parseVHDInfo(line, extractUuidFunction):
     return vhdInfo
 
 def _getVHDParentNoCheck(path):
-    cmd = ["vhd-util", "read", "-p", "-n", "%s" % path]
+    cmd = [VHD_UTIL, "read", "-p", "-n", "%s" % path]
     text = util.pread(cmd)
     util.SMlog(text)
     for line in text.split('\n'):
