commit 057e15bb39eea1de2e40967d8967477cf3c98e3e
Author: Bob Ball <bob.ball@citrix.com>
Date:   Wed Jan 7 09:44:20 2015 +0000

    xcp-sm-pylint-fix.patch

diff --git a/drivers/cleanup.py b/drivers/cleanup.py
index 31ccb50..66e9197 100755
--- a/drivers/cleanup.py
+++ b/drivers/cleanup.py
@@ -115,7 +115,7 @@ class Util:
         (stdout, stderr) = proc.communicate(inputtext)
         stdout = str(stdout)
         stderr = str(stderr)
-        rc = proc.returncode
+        rc = proc.returncode # pylint: disable=E1101
         if log:
             Util.log("`%s`: %s" % (args, rc))
         if type(expectedRC) != type([]):
diff --git a/drivers/util.py b/drivers/util.py
index caeca96..06f55a4 100755
--- a/drivers/util.py
+++ b/drivers/util.py
@@ -1418,6 +1418,7 @@ class extractXVA:
     #   returns filename, checksum content. Returns filename, '' in case  
     #   of checksum file missing. e.g. ova.xml
     def getTuple(self):
+        # pylint: disable=E1101
         zerod_record = 0
         ret_f_name = ''
         ret_base_f_name = ''
diff --git a/tests/pylintrc b/tests/pylintrc
index 29649ce..eb370ef 100644
--- a/tests/pylintrc
+++ b/tests/pylintrc
@@ -40,7 +40,7 @@ load-plugins=
 # append the F0401 code at the end of the "disable" config variable to temporarily
 # disable it from complaining missing libraries and get some useful information
 # from a partial run at least. 
-disable=design,rpython,newstyle,R,C,W,I
+disable=design,rpython,newstyle,R,C,W,I,F0401
 #,F0401
 
 
